clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/CallbackMap.java

```diff
@@ -9,21 +9,23 @@
 import java.util.concurrent.ConcurrentHashMap;
 
 public class CallbackMap {
-    private ConcurrentHashMap<String, List<ActionBase>> handlers = new ConcurrentHashMap<>();
+    private ConcurrentHashMap<String, List<InvocationHandler>> handlers = new ConcurrentHashMap<>();
 
-    public void put(String target, ActionBase action) {
+    public InvocationHandler put(String target, ActionBase action, ArrayList<Class<?>> classes) {
+        InvocationHandler handler = new InvocationHandler(action, classes);
```

Comment:
I suggest passing `classes` through [`Collection.unmodifiableList`](https://docs.oracle.com/javase/1.5.0/docs/api/java/util/Collections.html#unmodifiableList(java.util.List)) to make sure we can't accidentally modify it when we didn't expect to be able to.

Reply:
Java doesn't seem to have an equivalent for `IReadOnlyList` but `unmodifiableList` is what came up when I googled read only lists ;)